### PR TITLE
Validate several ids in constructors

### PR DESCRIPTION
**Pull request by jmartinesp**
These include:

* UserId & SessionId.
* EventId & ThreadId.
* RoomId & SpaceId.

### CODE REVIEW

File: features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/model/RoomListRoomSummaryProvider.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ open class RoomListRoomSummaryProvider : PreviewParameterProvider<RoomListRoomSu
 
 fun aRoomListRoomSummary() = RoomListRoomSummary(
     id = "!roomId",
-    roomId = RoomId("!roomId"),
+    roomId = RoomId("!roomId:matrix.org"),
```
Comment (bmarty) [inline features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/model/RoomListRoomSummaryProvider.kt:37]: `domain`

File: libraries/matrix/test/src/main/kotlin/io/element/android/libraries/matrix/test/TestData.kt [hunk 2]
```diff
@@ -29,8 +29,8 @@ const val A_PASSWORD = "password"
 
 val A_USER_ID = UserId("@alice:server.org")
 val A_USER_ID_2 = UserId("@bob:server.org")
-val A_SESSION_ID = SessionId(A_USER_ID.value)
-val A_SESSION_ID_2 = SessionId(A_USER_ID_2.value)
+val A_SESSION_ID = A_USER_ID
+val A_SESSION_ID_2 = A_USER_ID_2
```
Comment (bmarty) [inline libraries/matrix/test/src/main/kotlin/io/element/android/libraries/matrix/test/TestData.kt:33]: Either we remove the type `SessionId`, and we only use `UserId`, or I would prefer the previous version of the code.

File: features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/model/RoomListRoomSummaryPlaceholders.kt [hunk 3]
```diff
@@ -34,7 +34,7 @@ object RoomListRoomSummaryPlaceholders {
     fun createFakeList(size: Int): List<RoomListRoomSummary> {
         return mutableListOf<RoomListRoomSummary>().apply {
             repeat(size) {
-                add(create("\$fakeRoom$it"))
+                add(create("!fakeRoom:$it"))
```
[the remark this answers is not in the document]
Comment (jmartinesp) [reply features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/model/RoomListRoomSummaryPlaceholders.kt:37]: Yes, it's probably better to append it to the room part of the id instead. I changed it like this as a test and forgot to revert it 😅 .

File: libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/core/EventId.kt [hunk 4]
```diff
@@ -20,10 +20,12 @@ import io.element.android.libraries.matrix.api.BuildConfig
...
+        if (BuildConfig.DEBUG && !MatrixPatterns.isEventId(value)) {
+            error("`$value` is not a valid event id.\nExample event id: `\$Rqnc-F-dvnEYJTyHq_iKxU2bZ1CI92-kuZq3a5lr5Zg`.")
+        }
+    }
 
-fun String.asEventId() = if (BuildConfig.DEBUG && !MatrixPatterns.isEventId(this)) {
-    error("`$this` is not a valid event Id")
-} else {
-    EventId(this)
+    override fun toString(): String = value
```
[the remark this answers is not in the document]
Comment (jmartinesp) [reply libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/core/EventId.kt:30]: > If we keep this, we could do some cleanup where .value is used, for instance [here](https://github.com/vector-im/element-x-android/blob/develop/libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/factories/PendingIntentFactory.kt#L79)

String interpolation was the exact reason I did this change. For me, it makes it easier to use in logs and UI, but it's something subjective.

**Pull request merged**